activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,69 @@
+require 'active_support/time'
+
+module ActiveSupport
+  class Claims # :nodoc:
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires_at
+    UNIVERSAL_PURPOSE = 'universal'
+
+    def initialize(payload:, **options)
+      @payload = payload
+      @purpose = self.class.pick_purpose(options)
+      @expires_at = pick_expiration(options)
+    end
+
+    class << self
+      attr_accessor :expires_in
+
+      def verify!(claims, options = {})
```

Comment:
>>Sure, we can still get at them in the verify methods, but the nice thing about having that stuff contained in validate is that anything we add there will automatically improve the verify methods. This means the verify methods don't have to change.

Ok, I am also inclined to using ```validate``` now. Not having to change ```verify``` and ```verify!``` would be better. :+1: 





>>check_purpose? doesn't capture it for me. What does it mean for a purpose to be checked? How does it pass and how does it fail? same_purpose? is clearer IMO.

By ```check_purpose```, I meant that in the ```verify``` method of ```Claims```, we are basically checking if the purpose we have in ````claims``` matches with the one in the ```options```(entered in ```verify``` of ```MessageVerifier```). If the purposes don't match, then we are raising an exception. I don't think there is any harm in going with either of these :smile:.  

>>I also think unexpired? is better than not_expired?. Though I think it would be better to find a word that focuses on the freshness - you know instead of saying unexpired what's the "positive" way of saying the token is still alive.

That's an interesting thought.